plugin/resolve/DirectorySwitcherPlugin.js

```diff
@@ -0,0 +1,49 @@
+const path = require('path');
+const glob = require('glob');
+
+/**
+ * This plugin is intended to facilitate switching between a source directory
+ * and a transpiled distribution directory.
+ */
+class DirectorySwitcherPlugin {
+  constructor({
+    shouldSwitch = true, source = 'src', distribution = 'lib', rootDirectories = [process.cwd()],
```

Comment:
Do we expect others to use these plugins directly? I was wondering since we added fallbak values but defined them in the dev-site's webpack config. If so, should we move to `config/webpack/plugins`?

Also, can we add docs on the plugin configuration options in `plugin/resolve/README.md`?